UI/window-basic-main.cpp

```diff
@@ -1603,6 +1619,301 @@ void OBSBasic::UpdateMultiviewProjectorMenu()
+
+	// Calculate output file path
+	std::string strOutputFilePath =
+		config_get_string(Config(), "ScreenShots", "FilePath");
+
+	char lastChar = strOutputFilePath.back();
+	if (lastChar != '/' && lastChar != '\\')
+		strOutputFilePath += "/";
+
+	// TODO: Take this out to configuration
+	strOutputFilePath += GenerateSpecifiedFilename("png", noSpace, "%CCYY-%MM-%DD %hh-%mm-%ss");
+
+	replace(strOutputFilePath.begin(), strOutputFilePath.end(), '\\', '/');
+
+	// Get & ensure directory exists
+	size_t last = strOutputFilePath.rfind('/');
+	if (last == string::npos)
+		return;
+
+	string directory = strOutputFilePath.substr(0, last);
+	os_mkdirs(directory.c_str());
+
+	// Save screenshot to specified file path
+	bool success = SaveScreenshotToFile(strOutputFilePath.c_str());
+
+	if (success)
+	{
+		// TODO: Audio/visual feedback
+	}
+}
+
+///
+// Use libavcodec to save an input RGBA buffer to PNG
+//
+// @param outputFilePath		PNG file path to write to
+// @param width				image_data_ptr buffer width (pixels)
+// @param height			image_data_ptr buffer height (pixels)
+// @param image_data_ptr		RGBA buffer containing image data
+// @param image_data_ptr_buffer_length	image_data_ptr data length
+// @param image_data_linesize		line size (pitch)
+//
+// @returns				true on success, otherwise false
+//
+// @note				used by OBSBasic::SaveScreenshotToFile
+//
+static bool SaveRGBABufferToPNGFile(
+	const char*	outputFilePath,
+	uint32_t	width,
+	uint32_t	height,
+	uint8_t*	image_data_ptr,
+	size_t		image_data_buffer_length,
+	uint32_t	image_data_linesize)
+{
+	UNUSED_PARAMETER(image_data_buffer_length);
+
+	bool success = false;
+
+	AVCodec*		codec = NULL;
+	AVCodecContext*		codec_context = NULL;
+	AVFrame*		frame = NULL;
+	struct SwsContext*	swsc = NULL;
+	AVPacket		pkt;
+	int			avcodec_encode_video2_got_output = 0;
+
+	if (image_data_ptr == NULL)
+		goto err_no_image_data;
+
+	// Write PNG output using libavcodec
+	codec =
+		avcodec_find_encoder(AV_CODEC_ID_PNG);
+
+	if (codec == NULL)
+		goto err_png_codec_not_found;
+
+	codec_context =
+		avcodec_alloc_context3(codec);
+
+	if (codec_context == NULL)
+		goto err_png_encoder_context_alloc;
+
+	codec_context->width = width;
+	codec_context->height = height;
+	codec_context->pix_fmt = AV_PIX_FMT_RGB24;
+	codec_context->time_base.num = 1;
+	codec_context->time_base.den = 1000;
+	codec_context->codec_id = codec->id;
+
+	if (avcodec_open2(codec_context, codec, NULL) != 0)
+		goto err_png_encoder_open;
+
+	frame =
+		av_frame_alloc();
+
+	if (frame == NULL)
+		goto err_av_frame_alloc;
+
+	frame->pts = 1;
+	frame->format = AV_PIX_FMT_RGBA;
+	frame->width = width;
+	frame->height = height;
+
+	if (av_image_alloc(
+		&frame->data[0],
+		&frame->linesize[0],
+		codec_context->width,
+		codec_context->height,
+		codec_context->pix_fmt, 32) < 0)
+	{
+		goto err_av_image_alloc;
+	}
+
+	// Copy image data, converting RGBA to
+	// image format expected by PNG encoder
+	swsc =
+		sws_getContext(
+			frame->width,
```

Comment:
So the overall formatting, while satisfying the line size limit just feels off a bit (sorry, and not sure if @jp9000 would mention this in his review), i.e. it's density is really low; I'd suggest something like
```
	swsc = sws_getContext(frame->width, frame->height, AV_PIX_FMT_RGBA,
			frame->width, frame->height, codec_context->pix_fmt,
			0, NULL, NULL, NULL);
```
semi-perhaps-logical grouping of parameters, so function calls don't span too many lines

Reply:
I disagree. Breaking args into one-per-line leaves space for comments.

Reply:
Sure, but there are no comments, so it's wasted space at this point